### ISSUE TITLE
TypeScript errors do not interpolate values correctly (name out of scope?)

### ISSUE THREAD

**Issue by markfuller** [lyraproj/lyra#192]
**Describe the bug**

When developing workflows, and making mistakes, the Error that is thrown does not seem to interpolate the error values (I think because of `ReferenceError: name is not defined` - more below)

ASIDE: I don't know what the problem with the workflow is, but I suspect it could be [this line](https://github.com/markfuller/lyra/blob/ba2b7585dd00c18d32168e57ef935a05799ff83b/examples/ts-samples/src/sample_ts.ts#L11)?

**To Reproduce**
Get https://github.com/markfuller/lyra/tree/ts-example and then apply the sample_ts workflow, as follows:

```sh
make
cd examples/ts-samples
npm install
cd ../../
# verify the dist folder has been created
build/lyra apply sample_ts --debug
```

Gets output

```
2019-03-21T16:30:17.775Z [DEBUG] lyra: GetExternal: intId=lyra://puppet.com/sample_ts/person?hid=Example%3A%3APerson&rt=Example%3A%3APerson extId=<nil>
2019-03-21T16:30:17.778Z [DEBUG] lyra.node: /Users/Mark/code/anotherlyra/lyra/examples/ts-samples/node_modules/lyra-workflow/dist/servicesdk/Service.js:149
2019-03-21T16:30:17.778Z [DEBUG] lyra.node:                     throw Error(`state ${name} cannot be produced. Missing input parameter ${pn}`);
2019-03-21T16:30:17.778Z [DEBUG] lyra.node:                                          ^
2019-03-21T16:30:17.778Z [DEBUG] lyra.node:
2019-03-21T16:30:17.778Z [DEBUG] lyra.node: ReferenceError: name is not defined
2019-03-21T16:30:17.778Z [DEBUG] lyra.node:     at Function.callWithNamedArgs (/Users/Mark/code/anotherlyra/lyra/examples/ts-samples/node_modules/lyra-workflow/dist/servicesdk/Service.js:149:42)
2019-03-21T16:30:17.778Z [DEBUG] lyra.node:     at Service.state (/Users/Mark/code/anotherlyra/lyra/examples/ts-samples/node_modules/lyra-workflow/dist/servicesdk/Service.js:117:24)
2019-03-21T16:30:17.778Z [DEBUG] lyra.node:     at Object.state (/Users/Mark/code/anotherlyra/lyra/examples/ts-samples/node_modules/lyra-workflow/dist/servicesdk/Service.js:58:66)
2019-03-21T16:30:17.778Z [DEBUG] lyra.node:     at /Users/Mark/code/anotherlyra/lyra/examples/ts-samples/node_modules/grpc/src/server.js:590:13
2019-03-21T16:30:17.789Z [DEBUG] lyra: plugin process exited: path=/usr/local/bin/node pid=21580 error="exit status 1"
rpc error: code = Unavailable desc = transport is closing
```

See additional context for workaround steps - e.g. populating `name`.

**Expected behavior**
Something like `state blah cannot be produced. Missing input paramter foo`

**Logs**
Above

**Environment (please complete the following information):**
```sh
➜  lyra git:(ts-example) ✗ node --version
v11.12.0
➜  lyra git:(ts-example) ✗ tsc --version
Version 3.3.4000
```

**Additional context**
 
Source code at https://github.com/lyraproj/ts-workflow/blob/558dae86675fc43af5b3ede91d88fe536319f7df/lib/servicesdk/Service.ts#L181

Changing line (in generated `examples/ts-samples/node_modules/lyra-workflow/dist/servicesdk/Service.js` file)  to 

```js
                    throw "state" + name + " cannot be produced. Missing input parameter " + pn;
```

gets

```sh
2019-03-21T16:50:16.180Z [DEBUG] lyra.node: ReferenceError: name is not defined
```

Hence can work around like this: 

```js
                    var name = "ANY_NAME"
                    throw Error(`state ${name} cannot be produced. Missing input parameter ${pn} EOM`);
```

getting

```sh
2019-03-21T16:54:18.831Z [DEBUG] lyra.node: Error: state ANY_NAME cannot be produced. Missing input parameter  EOM
```

**Comment by markfuller** [comment]
Could be related to this https://github.com/Microsoft/TypeScript/issues/9850

(usage of name out of scope fails at runtime but not at compile time)

**Comment by markfuller** [comment]
https://github.com/lyraproj/ts-workflow/pull/2

**Comment by markfuller** [comment]
Fixed by #194
